PayloadPkg/Library/PayloadEntryLib/PayloadEntryLib.c

```diff
@@ -78,20 +80,32 @@ PayloadInit (
   GetPayloadReservedRamRegion (&RsvdBase, &RsvdSize);
   ASSERT ((RsvdBase & EFI_PAGE_MASK) == 0);
 
+  if (FeaturePcdGet (PcdDmaProtectionEnabled)) {
+    DmaSize = ALIGN_UP (PcdGet32 (PcdDmaBufferSize), EFI_PAGE_SIZE);
+  } else {
+    DmaSize = 0;
+  }
+  DmaBase  = (UINT32)RsvdBase - DmaSize;
+  DmaBase  = ALIGN_DOWN(DmaBase, PcdGet32 (PcdDmaBufferAlignment));
+

```

Comment:
do we need update SplitMemroyMap() in the Stage2Support.c to add DMA memory range in the memory map, so that all the payload could get DMA memory range from memory map?

Reply:
DMA memory is free memory to OS and we don't reserve it.  So if we report it in memory type, which type to use ?
From my perspective, payload should not need to know where is the DMA memory base/size since they will just use AllocateRuntimePages/Pools to get it.

Reply:
this is my thought to avoid to build another HOB to payload:
Memory map was updated in SBL:
If Stage2 directly boots OS, the DMA size should be set to 0.
If using UEFI payload to boot OS, the SblParseLib could find DMA memory range from memory map to use for DMA later. And convert to free memory after use.
If using Osloader, it could do similar with UEFI payload.

Reply:
After chat, we know UEFI payload will use page table to protect DMA which is different SBL methods. so let's use this method firstly in SBL.